python_modules/dagster/dagster/_core/storage/event_log/sql_event_log.py

```diff
@@ -2639,20 +2639,61 @@ def get_asset_check_execution_history(
         with self.index_connection() as conn:
             rows = conn.execute(query).fetchall()
 
-        return [
-            AssetCheckExecutionRecord(
-                id=cast(int, row["id"]),
-                run_id=cast(str, row["run_id"]),
-                status=AssetCheckExecutionRecordStatus(row["execution_status"]),
-                evaluation_event=(
-                    deserialize_value(cast(str, row["evaluation_event"]), EventLogEntry)
-                    if row["evaluation_event"]
-                    else None
-                ),
-                create_timestamp=datetime_as_float(cast(datetime, row["create_timestamp"])),
+        return [AssetCheckExecutionRecord.from_db_row(row) for row in rows]
+
+    def get_latest_asset_check_execution_by_key(self, asset_check_keys: Sequence[AssetCheckKey]):
```

Comment:
I think a second query here makes sense because
- in the majority of cases we only care about the latest execution, it's only the check history modal where we need the other query
- limit is weird, maybe would have to be limit_per_check? but we don't currently need that